code/obj/machinery/door/airlock.dm

```diff
@@ -2155,4 +2176,50 @@ TYPEINFO(/obj/machinery/door/airlock)
 						src.attach_signaler(which_wire+1, usr)
 						. = TRUE
 
+/obj/machinery/door/airlock/proc/user_demagging(mob/user, obj/item/current_tool)
+	. = FALSE
+
+	if(!user.traitHolder.hasTrait("training_engineer")) // If user doesn't have Engineer Training trait, simply return FALSE
+		return
+
+	switch(src.demag_status)
+		if(UNSCREW_STAGE)
+			if(isscrewingtool(current_tool))
+				boutput(user, "You unscrew the maintenance panel cover and remove it.")
+				playsound(src, 'sound/items/Screwdriver.ogg', 40)
+				src.demag_status = CUT_WIRE_STAGE
+				return TRUE
+
+		if(CUT_WIRE_STAGE)
+			if(issnippingtool(current_tool))
+				boutput(user, "You cut away and discard the burnt wiring.")
+				playsound(src, 'sound/items/Wirecutter.ogg', 40)
+				src.demag_status = INSERT_WIRE_STAGE
+				return TRUE
+
+		if(INSERT_WIRE_STAGE)
+			if(istype(current_tool, /obj/item/cable_coil) && current_tool.amount > 5) // If the user is using a cable coil on the airlock and that cable coil is larger than 5...
+				boutput(user, "You start inserting new wiring in the airlock...")
+				actions.start(new /datum/action/bar/icon/hitthingwithitem(src, user, current_tool, null, src, 10 SECONDS, /obj/machinery/door/airlock/proc/finish_inserting_wire,
+				list(current_tool), current_tool.icon, current_tool.icon_state, "You finish re-wiring the insides of the airlock."), user)
+				return TRUE
+
+		if(PULSE_STAGE)
+			if(ispulsingtool(current_tool))
+				boutput(user, "You restart the airlock's mainboard, and the airlock flickers back into functionality!")
+				src.demag(user)
+				src.demag_status = UNSCREW_STAGE // Reset the status for if the airlock gets emagged again
+				src.panel_open = TRUE // Because we *did* open the cover as the first step of de-emagging it.
+				return TRUE
+
+/obj/machinery/door/airlock/proc/finish_inserting_wire(obj/item/wires) // Called after the actionbar of inserting wiring in the airlock.
+	playsound(src, 'sound/items/Deconstruct.ogg', 40) // It's the most fitting sound, don't blame me for using it
+	wires.amount -= 5
```

Comment:
not the proper way to use wires, should be using the proc

Reply:
mb, i read some of the other code surrounding it and assumed this was the best way to do that.